Add markdown readme for System.Configuration.ConfigurationManager (#72772)
dotnet/runtime: src/libraries/System.Configuration.ConfigurationManager/src/README.md

## About

> Provides types that support using XML configuration files (`app.config`). Only use this package when migrating the existing .NET Framework code that already uses XML configuration. When writing a new code, use [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) instead.

Reviewer: ```suggestion
Provides types that support using XML configuration files (`app.config`). This package exists only to support migrating existing .NET Framework code that already uses System.Configuration. When writing new code, use another configuration system instead, such as [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/).
```

> Provides types that support using XML configuration files (`app.config`). This package is mostly useful to assist in porting of the existing .NET Framework code that depends on XML configuration. For a modern configuration API, see [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/).

Reviewer: Should we make the deprecation statement stronger? This package only exists to support migration. There is no reason to write new code against this API unless it is in maintenance of code migrated from .NET Framework that already uses it.

Author: Updated
